src/rust/protocol/src/memcache/wire/response/mod.rs

```diff
@@ -251,20 +252,42 @@ impl Compose for MemcacheResponse {
                     dst.write_all(self.result.as_bytes());
                     dst.write_all(&*entry.key);
 
-                    if cas {
-                        write!(
-                            dst,
-                            " {} {} {}",
-                            entry.flags,
-                            value.len(),
-                            entry.cas.unwrap_or(0)
-                        )
-                    } else {
-                        write!(dst, " {} {}", entry.flags, value.len())
+                    match value {
+                        Value::Bytes(value) => {
+                            if cas {
+                                write!(
+                                    dst,
+                                    " {} {} {}",
+                                    entry.flags,
+                                    value.len(),
+                                    entry.cas.unwrap_or(0)
+                                )
+                            } else {
+                                write!(dst, " {} {}", entry.flags, value.len())
+                            };
+                            dst.write_all(CRLF.as_bytes());
+                            dst.write_all(value);
+                        }
+                        Value::U64(value) => {
+                            let value_string = format!("{}", value);
+                            let value = value_string.as_bytes();
+                            if cas {
+                                write!(
+                                    dst,
+                                    " {} {} {}",
+                                    entry.flags,
+                                    value.len(),
+                                    entry.cas.unwrap_or(0)
+                                )
+                            } else {
+                                write!(dst, " {} {}", entry.flags, value.len())
+                            };
+                            dst.write_all(CRLF.as_bytes());
+                            dst.write_all(value);
+                        }
+                        Value::I64(_) => {}
```

Comment:
not implemented?

Reply:
Per discussion, I'm removing the signed i64 type until we have a concrete use-case but leaving the infrastructure for supporting multiple classes of typed values so we can add more types in the future.